src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,60 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+import {
+  getLinkDataCy,
+  getLinkHref,
+  getLinkTextContent,
+  handleClickPagination,
+} from './pagination-utils';
+
+export interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesAmount = Math.ceil(total / perPage);
+
+  return (
+    <ul
+      className="pagination"
+      onClick={(event: React.MouseEvent<HTMLUListElement>) =>
+        handleClickPagination(event, currentPage, pagesAmount, onPageChange)
+      }
+    >
+      {[...Array(pagesAmount + 2)].map((_item, index) => {
```

Comment:
Using `Math.random()` for keys in React is not recommended as it will generate a new key on every render, leading to the loss of the component state and unnecessary re-renders. Instead, use a unique and consistent identifier such as the `index` or a combination of `index` and another string.